### PR TITLE
APP-374: Add events demo for android

### PR DESCRIPTION
**Pull request by antonshkurenko**

### CODE REVIEW

**Comment by Slash32** [comment]
@kaa-leeroy test this fully please

**Comment by Slash32** [comment]
@kaa-leeroy test this fully please

**Comment by antonshkurenko** [comment]
@kaa-leeroy test this fully please

**Comment by Slash32** [comment]
@kaa-leeroy test this fully please

**Comment by Slash32** [comment]
@kaa-leeroy test this fully please

**Comment by Slash32** [comment]
@kaa-leeroy test this fully please

**Review by Slash32 — CHANGES_REQUESTED** [review]
Please drop the icon.png file since it will conflict with the other one added in a separate PR.

**Comment by Slash32** [comment]
@kaa-leeroy test this fully please

**Review by Slash32 — CHANGES_REQUESTED** [review]
The target merge branch is release-2.1 which has not been created yet.

File: eventdemo/source/android/app/src/main/java/org/kaaproject/kaa/demo/events/utils/KaaChatManager.java [hunk 1]
```diff
@@ -0,0 +1,285 @@
...
+                @Override
+                public void onStopped() {
+                    toast("Kaa client stopped");
+                }
+            }, true);
+
+            //Start the Kaa client and connect it to the Kaa server.
+            mKaaClient.start();
+
+            startupLatch.await();
```
Comment (abohomol) [inline eventdemo/source/android/app/src/main/java/org/kaaproject/kaa/demo/events/utils/KaaChatManager.java:108]: The endpoint SDK enables you to have event-driven flow via appropriate callback, so why do you make those calls synchronized?

File: eventdemo/source/android/app/src/main/java/org/kaaproject/kaa/demo/events/utils/KaaChatManager.java [hunk 2]
```diff
@@ -0,0 +1,285 @@
...
+                    } else {
+                        toast("Attaching current endpoint to user [ID=%s] FAILED.", userId);
+                        toast("Attach response: %s", response);
+                        toast("Events exchange will be NOT POSSIBLE.");
+                    }
+                    attachLatch.countDown();
+                }
+            });
+
+            attachLatch.await();
```
Comment (abohomol) [inline eventdemo/source/android/app/src/main/java/org/kaaproject/kaa/demo/events/utils/KaaChatManager.java:197]: Same as a previous one

File: eventdemo/source/android/app/src/main/java/org/kaaproject/kaa/demo/events/utils/KaaChatManager.java (hunk 1 above)
Comment (abohomol) [reply eventdemo/source/android/app/src/main/java/org/kaaproject/kaa/demo/events/utils/KaaChatManager.java:108]: It's also stopping EventDemoApp from being created while you're waiting for result.

File: eventdemo/source/android/app/src/main/java/org/kaaproject/kaa/demo/events/dialogfragments/InputTextDialogFragment.java [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+                    dismiss();
+                    break;
+            }
+        }
+    };
+
+    /**
+     * Create new instance of the fragment with args
+     *
+     * @param hint hint has a second function,
```
Comment (abohomol) [inline eventdemo/source/android/app/src/main/java/org/kaaproject/kaa/demo/events/dialogfragments/InputTextDialogFragment.java:68]: Although, the second function of anything is a bad practice it's event worse to give that to a hint of EditText.

File: eventdemo/source/android/app/src/main/java/org/kaaproject/kaa/demo/events/dialogfragments/InputTextDialogFragment.java [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+    public Dialog onCreateDialog(Bundle savedInstanceState) {
+        final Context context = getContext();
+
+        final View rootView = LayoutInflater.from(context).inflate(R.layout.dialog_input_text, null);
+        mInputText = (EditText) rootView.findViewById(R.id.input_text);
+
+        mInputText.setHint(getArguments().getString(ARGS_HINT));
+
+        final AlertDialog.Builder builder = new AlertDialog.Builder(context);
+        builder.setTitle("Input text");
```
Comment (abohomol) [inline eventdemo/source/android/app/src/main/java/org/kaaproject/kaa/demo/events/dialogfragments/InputTextDialogFragment.java:95]: Apart from demonstrating the best practices of endpoint SDK usage application should also demonstrate solid Android skills, so please replace all literals with string resources.

File: eventdemo/source/android/app/src/main/java/org/kaaproject/kaa/demo/events/dialogfragments/InputTextDialogFragment.java [hunk 5]
```diff
@@ -0,0 +1,134 @@
...
+import android.view.LayoutInflater;
+import android.view.View;
+import android.widget.EditText;
+
+import org.kaaproject.kaa.demo.events.R;
+
+/**
+ * Util class to input text in dialog
+ */
+public final class InputTextDialogFragment extends DialogFragment {
```
Comment (abohomol) [inline eventdemo/source/android/app/src/main/java/org/kaaproject/kaa/demo/events/dialogfragments/InputTextDialogFragment.java:36]: Isn't that cheaper to create a dialog using the "AlertDialog.Builder" and set EditText to its view?

**Review by abohomol — CHANGES_REQUESTED** [review]
I recommend to make certain changes before merging.

**Pull request closed without merging**